hadoop-ozone/recon/src/main/java/org/apache/hadoop/ozone/recon/codec/NSSummaryCodec.java

```diff
@@ -117,6 +120,14 @@ public NSSummary fromPersistedFormat(byte[] rawData) throws IOException {
     assert (bytesRead == strLen);
     String dirName = stringCodec.fromPersistedFormat(buffer);
     res.setDirName(dirName);
+    long parentId = in.readLong();
+    if (parentId == 0) {
+      // Set the parent ID to -1 to indicate that it is old data from
+      // the cluster and the value has not yet been set.
+      res.setParentId(-1);
```

Comment:
parentId "0" and "-1" seems represent same case, no-exist parentId. do this have any other differentiation, if no, then can use only one.

Reply:
`Zero` represents it does not have a parent ID but the data is newly created. And `-1` signifies that this is an old data and will need a full reprocess trigger.